frappe/desk/query_report.py

```diff
@@ -486,8 +500,9 @@ def add_total_row(result, columns, meta=None):
 				else result[0][i]
 			)
 
-	for i in has_percent:
-		total_row[i] = flt(total_row[i]) / len(result)
+	if total_percentage_rows:
+		for i in has_percent:
+			total_row[i] = flt(total_row[i]) / total_percentage_rows
```

Comment:
The variable(`total_percentage_rows`) gets updated for each column but is used without the context of any column
https://github.com/frappe/frappe/pull/15950/files#diff-950a52ee4f8ec738a657c004db0a33bfaaa6474bc0d34ba4ec12d327e6bafbeaR484
Basically, it will have the count of non-null percentage rows in the last column